swap/src/protocol/alice/event_loop.rs

```diff
@@ -307,12 +308,27 @@ impl LatestRate for FixedRate {
     }
 }
 
-impl LatestRate for kraken::PriceUpdates {
+#[derive(Debug)]
+pub struct SpreadRate {
```

Comment:
🔧 I think `LatestRate` was a better name, because it expresses better what this is. This combines the latest price update with the `Rate` - the fact that the rate applies a spread is an implementation detail.

Reply:
`LatestRate` still exists, it is the trait name.

> This combines the latest price update with the Rate - the fact that the rate applies a spread is an implementation detail.

I'll try and think of a better name.